subsys/settings/src/settings_file.c

```diff
@@ -519,16 +517,51 @@ void settings_mount_file_backend(struct settings_file *cf)
 	settings_line_io_init(read_handler, write_handler, get_len_cb, 1);
 }
 
+static int mkdir_if_not_exists(const char *path)
+{
+	struct fs_dirent entry;
+	int err;
+
+	err = fs_stat(path, &entry);
```

Comment:
You do not have to check this, `fs_mkdir` is supposed to return `-EEXIST` if directory already exists.

Reply:
> But you have to check whether what does exist is directory because fs_stat will be successful for directory and for file if it exists.

Added check for that case.

> You do not have to check this, `fs_mkdir` is supposed to return `-EEXIST` if directory already exists, so you can just attempt to create it without checking.

True. However due to implemented check with `fs_stat` on FS entry type (whether it is a directory), it will be "cheaper" to do only one FS call in all subsequent system boots when directory already exists. So I left `fs_stat()` verification fist, then `fs_mkdir()` call if directory does not exists.